api/crl/crl.go

```diff
@@ -1,104 +1,90 @@
-// Package crl implements the HTTP handler for the crl commands.
+// Package crl implements the HTTP handler for the crl command.
 package crl
 
 import (
-	"crypto/rand"
-	"crypto/x509/pkix"
-	"encoding/json"
+	"crypto"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"os"
+
 	"github.com/cloudflare/cfssl/api"
+	"github.com/cloudflare/cfssl/certdb"
+	"github.com/cloudflare/cfssl/crl"
 	"github.com/cloudflare/cfssl/errors"
 	"github.com/cloudflare/cfssl/helpers"
 	"github.com/cloudflare/cfssl/log"
-	"io/ioutil"
-	"math/big"
-	"net/http"
-	"strconv"
-	"strings"
-	"time"
 )
 
-// This type is meant to be unmarshalled from JSON
-type jsonCRLRequest struct {
-	Certificate  string   `json:"certificate"`
-	SerialNumber []string `json:"serialNumber"`
-	PrivateKey   string   `json:"issuingKey"`
-	ExpiryTime   string   `json:"expireTime"`
+// A Handler accepts requests with a serial number parameter
+// and revokes
+type Handler struct {
+	dbAccessor certdb.Accessor
+	ca         *x509.Certificate
+	key        crypto.Signer
 }
 
-// Handle responds to requests for crl generation. It creates this crl
-// based off of the given certificate, serial numbers, and private key
-func crlHandler(w http.ResponseWriter, r *http.Request) error {
-
-	var revokedCerts []pkix.RevokedCertificate
-	var oneWeek = time.Duration(604800) * time.Second
-	var newExpiryTime = time.Now()
-
-	body, err := ioutil.ReadAll(r.Body)
+// NewHandler returns a new http.Handler that handles a revoke request.
+func NewHandler(dbAccessor certdb.Accessor, caPath string, caKeyPath string) (http.Handler, error) {
+	ca, err := ioutil.ReadFile(caPath)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	r.Body.Close()
-
-	req := &jsonCRLRequest{}
 
-	err = json.Unmarshal(body, req)
+	caKey, err := ioutil.ReadFile(caKeyPath)
 	if err != nil {
-		log.Error(err)
+		return nil, errors.Wrap(errors.CertificateError, errors.ReadFailed, err)
 	}
 
-	if req.ExpiryTime != "" {
-		expiryTime := strings.TrimSpace(req.ExpiryTime)
-		expiryInt, err := strconv.ParseInt(expiryTime, 0, 32)
-		if err != nil {
-			return err
-		}
-
-		newExpiryTime = time.Now().Add((time.Duration(expiryInt) * time.Second))
+	// Parse the PEM encoded certificate
+	issuerCert, err := helpers.ParseCertificatePEM(ca)
+	if err != nil {
+		return nil, err
 	}
 
-	if req.ExpiryTime == "" {
-		newExpiryTime = time.Now().Add(oneWeek)
+	strPassword := os.Getenv("CFSSL_CA_PK_PASSWORD")
```

Comment:
I agree - although, the very same environment variable is used in other commands (old crl now gencrl, local - https://github.com/cloudflare/cfssl/search?utf8=%E2%9C%93&q=CFSSL_CA_PK_PASSWORD), and apparently there's not a place where these environment variables are documented now. Let me know what you think, I might add something here or leave it to a future MR.